src/tmpfiles/tmpfiles.c

```diff
@@ -967,17 +967,17 @@ static int path_open_parent_safe(const char *path) {
         int r, fd;
 
         if (!path_is_normalized(path))
-                return log_error_errno(SYNTHETIC_ERRNO(EINVAL), "Failed to open parent of '%s': path not normalized.", path);
+                return log_debug_errno(SYNTHETIC_ERRNO(EINVAL), "Failed to open parent of '%s': path not normalized.", path);
 
         r = path_extract_directory(path, &dn);
         if (r < 0)
-                return log_error_errno(r, "Unable to determine parent directory of '%s': %m", path);
+                return log_debug_errno(r, "Unable to determine parent directory of '%s': %m", path);
 
         r = chase_symlinks(dn, arg_root, CHASE_SAFE|CHASE_WARN, NULL, &fd);
         if (r == -ENOLINK) /* Unsafe symlink: already covered by CHASE_WARN */
                 return r;
         if (r < 0)
-                return log_error_errno(r, "Failed to open path '%s': %m", dn);
+                return log_debug_errno(r, "Failed to open path '%s': %m", dn);
 
         return fd;
```

Comment:
hmm, so this doesn't look quite right. chase_symlinks() still has CHASE_WARN, and thus will log at higher log levels.

I think it would be better to move the allow_failure bool into a parameter for path_open_parent_safe(), so that you can toggle CHASE_WARN on that.

it's also shorter I guess because the callers don't have to generate a log message each

Reply:
I don't see anything from that though? This is about likely-to-happen errors, like having a read-only filesystem. Weird or unexpected failures are fine to be logged at error level.

Reply:
Well, but the fact that chase_symlink() is sometimes logging about errors here should not be leaked further up the stack. chase_symlink() returns a documented error code (ENOLINK) in case it logged already about that, and doesn't log at all about the rest. If you just propagate this up here, you will end up with duplicate logging in the ENOLINK case.

So, pleast just move the logging into path_open_parent_safe(), but make it configurable via a boolean or flag parameter. it's much simpler and causes much less copying around of the same code everywhere, because your current patch adds the same 5 lines of log call to every invocation of path_open_parent_safe(), and that really should be inside of path_open_parent_safe() instead.

Reply:
fine, updated